vm/src/stdlib/builtins.rs

```diff
@@ -248,11 +248,35 @@ mod builtins {
     #[cfg(feature = "rustpython-compiler")]
     #[pyfunction]
     fn eval(
-        source: Either<PyStrRef, PyRef<crate::builtins::PyCode>>,
+        source: Either<
+            Either<PyStrRef, crate::builtins::PyBytesRef>,
+            PyRef<crate::builtins::PyCode>,
+        >,
```

Comment:
is this exactly `str` or `bytes` or `ArgStrOrBytesLike`?
`ArgStrOrBytesLike` accepts more types like `bytearray`

Reply:
When I run rustpython with `ArgStrOrBytesLike`, it seems very suit for this case. `string, bytes or code` is just error message and the `eval` also receives `bytearray` `Buffer` 🤔 . So I'll push a commit apply `ArgStrOrBytesLike`. Thanks! 🙏🏻